pkg/apis/autoscaling/v2beta2/conversion_test.go

```diff
@@ -0,0 +1,124 @@
+	tests := []struct {
+		name        string
+		internalHPA *autoscaling.HorizontalPodAutoscaler
+	}{
+		{
+			"Complete HPA with scale-up tolerance",
+			&autoscaling.HorizontalPodAutoscaler{
+				ObjectMeta: v1.ObjectMeta{
+					Name:        "hpa",
+					Namespace:   "hpa-ns",
+					Annotations: map[string]string{"key": "value"},
+				},
+				Spec: autoscaling.HorizontalPodAutoscalerSpec{
+					MinReplicas: ptr.To(int32(1)),
+					MaxReplicas: 3,
+					Metrics: []autoscaling.MetricSpec{
+						{
+							Type: autoscaling.ResourceMetricSourceType,
+							Resource: &autoscaling.ResourceMetricSource{
+								Name: api.ResourceCPU,
+								Target: autoscaling.MetricTarget{
+									Type:         autoscaling.AverageValueMetricType,
+									AverageValue: resource.NewMilliQuantity(300, resource.DecimalSI),
+								},
+							},
+						},
+					},
+					Behavior: &autoscaling.HorizontalPodAutoscalerBehavior{
+						ScaleUp: &autoscaling.HPAScalingRules{
+							Policies: []autoscaling.HPAScalingPolicy{
+								{
+									Type:          autoscaling.PodsScalingPolicy,
+									Value:         1,
+									PeriodSeconds: 2,
+								},
+							},
+							Tolerance: &tolerance1,
+						},
+					},
+				},
+			},
+		},
+		{
+			"Scale-down and scale-up tolerances",
+			&autoscaling.HorizontalPodAutoscaler{
+				Spec: autoscaling.HorizontalPodAutoscalerSpec{
+					MinReplicas: ptr.To(int32(1)),
+					MaxReplicas: 3,
+					Behavior: &autoscaling.HorizontalPodAutoscalerBehavior{
+						ScaleUp: &autoscaling.HPAScalingRules{
+							Tolerance: &tolerance1,
+						},
+						ScaleDown: &autoscaling.HPAScalingRules{
+							Tolerance: &tolerance2,
+						},
+					},
+				},
+			},
+		},
+		{
+			"Scale-down tolerance only",
+			&autoscaling.HorizontalPodAutoscaler{
+				Spec: autoscaling.HorizontalPodAutoscalerSpec{
+					MinReplicas: ptr.To(int32(1)),
+					MaxReplicas: 3,
+					Behavior: &autoscaling.HorizontalPodAutoscalerBehavior{
+						ScaleDown: &autoscaling.HPAScalingRules{
+							Tolerance: &tolerance2,
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			v2beta2HPA := &autoscalingv2beta2.HorizontalPodAutoscaler{}
+			if err := Convert_autoscaling_HorizontalPodAutoscaler_To_v2beta2_HorizontalPodAutoscaler(tt.internalHPA, v2beta2HPA, nil); err != nil {
+				t.Errorf("Convert_autoscaling_HorizontalPodAutoscaler_To_v2beta2_HorizontalPodAutoscaler() error = %v", err)
+			}
+			roundtripHpa := &autoscaling.HorizontalPodAutoscaler{}
+			if err := Convert_v2beta2_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler(v2beta2HPA, roundtripHpa, nil); err != nil {
+				t.Errorf("Convert_v2beta2_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler() error = %v", err)
+			}
+			if !apiequality.Semantic.DeepEqual(tt.internalHPA, roundtripHpa) {
+				t.Errorf("HPA is not equivalent after round-trip:  mismatch (-want +got):\n%s", cmp.Diff(tt.internalHPA, roundtripHpa))
```

Comment:
Style nit:

```suggestion
			roundtripHPA := &autoscaling.HorizontalPodAutoscaler{}
			if err := Convert_v2beta2_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler(v2beta2HPA, roundtripHPA, nil); err != nil {
				t.Errorf("Convert_v2beta2_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler() error = %v", err)
			}
			if !apiequality.Semantic.DeepEqual(tt.internalHPA, roundtripHPA) {
				t.Errorf("HPA is not equivalent after round-trip:  mismatch (-want +got):\n%s", cmp.Diff(tt.internalHPA, roundtripHPA))
```

To match the `v2beta2HPA` above

Reply:
Done.